--- origsrc/libmpc/mpcenc/mpcenc.h	2010-01-19 19:29:03.000000000 -0600
+++ src/libmpc/mpcenc/mpcenc.h	2010-01-19 19:50:51.607246600 -0600
@@ -43,6 +43,7 @@
 # include <windows.h>
 #elif defined __unix__  ||  defined __linux__  ||  defined __APPLE__
 # include <unistd.h>
+# include <fcntl.h>
 # include <locale.h>
 # include <langinfo.h>
 #else
@@ -128,10 +129,7 @@
 #elif defined __TURBOC__ || defined __BORLANDC__
 # define SETBINARY_IN(__fp)     (setmode   ( FILENO ((__fp)),  O_BINARY ), (__fp))
 # define SETBINARY_OUT(__fp)    (setmode   ( FILENO ((__fp)),  O_BINARY ), (__fp))
-#elif defined __CYGWIN__
-# define SETBINARY_IN(__fp)     (setmode   ( FILENO ((__fp)), _O_BINARY ), (__fp))
-# define SETBINARY_OUT(__fp)    (setmode   ( FILENO ((__fp)), _O_BINARY ), (__fp))
-#elif defined _WIN32
+#elif defined _WIN32 || defined __CYGWIN__
 # define SETBINARY_IN(__fp)     (_setmode  ( FILENO ((__fp)), _O_BINARY ), (__fp))
 # define SETBINARY_OUT(__fp)    (_setmode  ( FILENO ((__fp)), _O_BINARY ), (__fp))
 #elif defined _MSC_VER
